plugins/jenkins/lib/samson_jenkins/samson_plugin.rb

```diff
@@ -0,0 +1,33 @@
+module SamsonJenkins
+  class Engine < Rails::Engine
+  end
+end
+
+Samson::Hooks.view :stage_form, "samson_jenkins/fields"
+Samson::Hooks.view :deploys_header, "samson_jenkins/deploys_header"
+
+Samson::Hooks.callback :stage_permitted_params do
+  :jenkins_job_names
+end
+
+Samson::Hooks.callback :deploy_defined do
+  Deploy.class_eval do
+    has_many :jenkins_job
+  end
+end
+
+Samson::Hooks.callback :after_deploy do |stage, deploy|
+  if deploy.status == 'succeeded' && stage.jenkins_job_names?
+    stage.jenkins_job_names.to_s.split(/, ?/).map do |job_name|
+      job_id = Jenkins.new(job_name, deploy).build
+      attributes = {name: job_name, deploy_id: deploy.id}
+      if job_id.is_a?(Fixnum)
+        attributes[:jenkins_job_id] = job_id
+      else
+        attributes[:jenkins_job_status] = "CANCELED"
```

Comment:
I think you can use:

delegate :jenkins_job_status, :to => :'jenkins_job.status'

but jenkins_job.status / jenkins_job.try(:status) might even be
simpler/better

On Tue, Apr 21, 2015 at 10:42 AM, rupinder dhanoa <notifications@github.com>
wrote:

> In plugins/jenkins/lib/samson_jenkins/samson_plugin.rb
> <https://github.com/zendesk/samson/pull/310#discussion_r28803064>:
>
> > +
> > +Samson::Hooks.callback :deploy_defined do
> > +  Deploy.class_eval do
> > +    has_many :jenkins_job
> > +  end
> > +end
> > +
> > +Samson::Hooks.callback :after_deploy do |stage, deploy|
> > +  if deploy.status == 'succeeded' && stage.jenkins_job_names?
> > +    stage.jenkins_job_names.to_s.split(/, ?/).map do |job_name|
> > +      job_id = Jenkins.new(job_name, deploy).build
> > +      attributes = {name: job_name, deploy_id: deploy.id}
> > +      if job_id.is_a?(Fixnum)
> > +        attributes[:jenkins_job_id] = job_id
> > +      else
> > +        attributes[:jenkins_job_status] = "CANCELED"
>
> I was saying deploy. jenkins_job_status returns
> jenkins_job.jenkins_job_status
> thats to show on deploys page of a stage next to deploy status
> https://samson.zende.sk/projects/zendesk/deploys.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zendesk/samson/pull/310/files#r28803064>.
>